langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiResponsesStreamingChatModel.java

```diff
@@ -0,0 +1,972 @@
+                    tool.put(FIELD_TYPE, TYPE_FUNCTION);
+                    tool.put(FIELD_NAME, toolSpec.name());
+                    if (toolSpec.description() != null) {
+                        tool.put(FIELD_DESCRIPTION, toolSpec.description());
+                    }
+
+                    Map<String, Object> functionParameters = null;
+                    if (toolSpec.parameters() != null) {
+                        functionParameters = toMap(toolSpec.parameters(), strict);
+                    } else if (strict) {
+                        functionParameters = new HashMap<>();
+                        functionParameters.put(FIELD_TYPE, TYPE_OBJECT);
+                        functionParameters.put(FIELD_PROPERTIES, new HashMap<>());
+                        functionParameters.put(FIELD_ADDITIONAL_PROPERTIES, false);
+                    }
+
+                    if (functionParameters != null) {
+                        tool.put(FIELD_PARAMETERS, functionParameters);
+                    }
+
+                    if (strict) {
+                        tool.put(FIELD_STRICT, true);
+                    }
+
+                    tools.add(tool);
+                }
+                payload.put(FIELD_TOOLS, tools);
+
+                if (chatRequest.toolChoice() != null) {
+                    payload.put(FIELD_TOOL_CHOICE, toToolChoiceString(chatRequest.toolChoice()));
+                } else {
+                    payload.put(FIELD_TOOL_CHOICE, "auto");
+                }
+            }
+
+            var textConfig = toResponseTextConfig(chatRequest.responseFormat());
+            if (textVerbosity != null) {
+                if (textConfig == null) {
+                    textConfig = new HashMap<>();
+                }
+                textConfig.put(FIELD_TEXT_VERBOSITY, textVerbosity);
+            }
+            if (textConfig != null) {
+                payload.put(FIELD_TEXT, textConfig);
+            }
+
+            var requestBuilder = HttpRequest.builder()
+                    .url(baseUrl + "/responses")
+                    .method(HttpMethod.POST)
+                    .addHeader("Authorization", "Bearer " + apiKey)
+                    .addHeader("Content-Type", "application/json")
+                    .addHeader("Accept", "text/event-stream");
+
+            if (organizationId != null) {
+                requestBuilder.addHeader(OPENAI_ORGANIZATION_HEADER, organizationId);
+            }
+
+            String requestBody = OBJECT_MAPPER.writeValueAsString(payload);
+            var request = requestBuilder.body(requestBody).build();
+
+            var streamingHandle = new ResponsesStreamingHandle();
+            httpClient.execute(
+                    request,
+                    new DefaultServerSentEventParser(),
+                    new ResponsesApiEventListener(handler, streamingHandle));
+
+        } catch (Exception e) {
+            logger.error("Exception in doChat", e);
+            safeOnError(handler, e);
```

Comment:
```suggestion
            withLoggingExceptions(() -> handler.onError(e));
```
Please replace it everywhere `safeOnError` is ued